### DISCUSSION TITLE
Dreame l10 ultra. Automate your own custom cleaning

### DISCUSSION THREAD

**Discussion by ghost** [Tasshack/dreame-vacuum#698] [General]
Hello. I'm trying to do the following. The robot vacuums room number one, then mops, then vacuums room number 2, then mops. This can be achieved by properly setting the task shortcut, but there is a limit of only six tasks and the vacuum cleaner unnecessarily returns to the base (even with auto emptying, washing and drying options disabled). I was looking for an entity about the end of room cleaning, but I did not find the appropriate one. The l10 ultra status entity changes to the robot returning to the base, and the task status entity changes to completed, but when the robot reaches the base. Is there any solution?

**Comment by Tasshack** [comment]
> The l10 ultra status entity changes to the robot returning to the base

This is what you need to check for end of room cleaning.

**Comment by ghost** [comment]
this is not good. this entity changes for other reasons. (when will someone be needed to make the robot come back?) does it really make any difference about the tasks being thrown before returning to base?

**Comment by Tasshack** [comment]
If you manually command to vacuum to return to base then task_status will also change so you should compare status entity injuction with other entities like task_status, charging_status, state and vacuum entities. There is no way to know a cleaning job is completed just by checking a single entity, that is why there are too many status entities for combined vacuum states.

**Comment by ghost** [comment]
If there is no other way you need to change the entity. I have a problem here too. I'm thinking about listening to the 'sensor.l10_ultra_status' and 'sensor.l10_ultra_task_status' entities

if sensor.l10_ultra_status = 'returning' and sensor.l10_ultra_task_status = 'completed' then // the result is false

sensor.l10_ultra_task_status changes status to "completed" after 2, 4 seconds

Maybe I'll think of something. Thank you for your reply

**Reply by Tasshack** [reply]
> If there is no other way you need to change the entity.

This is how the device operates, what do you want me to do?
